Code/ItemRandoComponent.cs

```diff
@@ -0,0 +1,53 @@
+﻿using UnityEngine;
+
+namespace ArchipelagoRandomizer
+{
+	public abstract class ItemRandoComponent : MonoBehaviour
+	{
+		/// <summary>
+		/// Preloads any objects this component needs to reference
+		/// </summary>
+		public virtual void Preload(Preloader preloader)
```

Comment:
So I assume you wanted these to be optional, which is why you didn't use `abstract`?
I would think the better formatting for this would be something like

```cs
public virtual void Preload(Preloader preloader) { }
```

Cuts down on the excess space and makes it clear that it's supposed to be an empty body, rather than just an unfinished method. Same for the two below.

Reply:
Yeah I can change that. I prefer it that way as well, but I've been too lazy to update my auto-formatter rules, so when I save it changes it to how it is here